scio-grpc/src/test/scala/com/spotify/scio/grpc/GrpcBatchDoFnTest.scala

```diff
@@ -194,7 +194,7 @@ class GrpcBatchDoFnTest extends PipelineSpec with BeforeAndAfterAll {
     val input = (0 to 10).map { i =>
       ConcatRequestWithID
         .newBuilder()
-        .setRequestId("1")
+        .setRequestId(i.toString)
```

Comment:
But i think that shouldn't be the case .. 
The first test tests that given that all the inputs in the bundle return unique ids they have matched outputs in numbers
This test checks that if all inputs have the same exact id it would still match outputs in numbers 

The reasoning behind this check is that we do send a single request for the first element we find and use the response for all the matching inputs and i wanted to be sure that this behavior is maintained here.

Reply:
i would even go to say that this test is broken (the original one) 
If you follow the test thoroughly you would find that i have inputs with different values sharing the same request ids and when those go to the gRPC server and come back they have different output values ! so it makes perfect sense that they fail when they come back as the expectation i had was that for each input object i sent i would get a "different" output which doesn't make sense. 

i think we should change the test to just duplicate the inputs list and pass it to the scio pipline instead of manipulating the request ID

also .. i am insanely puzzled how this NEVER failed locally for me !

Reply:
Actually .. never mind .. i managed to force it to use bigger bundles and i think it has something to do with the number of CPUs on my machine "somehow" 

i created a LOT (2000) of duplicated input with a big batch size to properly test that scenario. 

just pushed the changes :)